nipoppy_cli/nipoppy/env.py

```diff
@@ -0,0 +1,9 @@
+"""Variable Definitions"""
+import os
+from typing import TypeVar
+
+StrOrPathLike = TypeVar("StrOrPathLike", str, os.PathLike)
```

Comment:
probably not needed yet but it may in the long run put all complex type definitions in a typing modules.

See: 
https://github.com/pandas-dev/pandas/blob/main/pandas/_typing.py
https://github.com/numpy/numpy/tree/main/numpy/_typing